alfalfa_worker/Dockerfile

```diff
@@ -150,17 +149,25 @@ RUN git clone --branch $SUNDIALS_TAG --depth 1 https://github.com/LLNL/sundials.
     && make install \
     && cd .. && rm -rf sundials sundials_build
 
-# install and upgrade setuptools because this is needed for symfit (used in E+ Python)
-#    For Assimulo -- numpy, Cython must match the version in pypoetry.toml
-RUN ln -sf /usr/bin/python3 /usr/bin/python \
-    && python3 -m pip install setuptools==65.3.0 numpy==1.21.6 Cython==0.29.28
-
 # This is required for Assimulo to build correctly with setuptools 60+
 ENV SETUPTOOLS_USE_DISTUTILS stdlib
 
+# Install poetry
+RUN pip3 install poetry==1.3.1
+
+WORKDIR /alfalfa
+
+# Copy and install poetry-based dependencies
+COPY pyproject.toml /alfalfa
+COPY poetry.lock /alfalfa
+
+# Install 'production' dependencies from poetry
+RUN poetry install --only main
```

Comment:
The only issue I see with moving this here is that if any dependencies are changed we have to rebuild pyfmi and assimulo

Reply:
I agree. But there was no way around this until we release our own pyfmi package and can just install it.

Reply:
```
=> [nrel/alfalfa-worker:latest base 20/25] RUN git clone --branch Assimulo-3.2.9 --depth 1 https://  124.3s
 => [nrel/alfalfa-worker:latest base 21/25] RUN git clone --branch PyFMI-2.9.5 --depth 1 https://git  176.8s
```
Looks like 5 minutes to rebuild when we change dependencies. But the poetry install in dev is faster now as it only installs the dev dependencies. Before we were creating two python virtual envs, one in the /home/build path and one in /alfalfa.